gallery/wdio.conf.js

```diff
@@ -0,0 +1,303 @@
+    // see also: https://webdriver.io/docs/frameworks.html
+    //
+    // Make sure you have the wdio adapter package for the specific framework installed
+    // before running any tests.
+    framework: 'mocha',
+    //
+    // The number of times to retry the entire specfile when it fails as a whole
+    // specFileRetries: 1,
+    //
+    // Whether or not retried specfiles should be retried immediately or deferred to the end of the queue
+    // specFileRetriesDeferred: false,
+    //
+    // Test reporter for stdout.
+    // The only one supported by default is 'dot'
+    // see also: https://webdriver.io/docs/dot-reporter.html
+    reporters: ['spec'],
+
+
+    
+    //
+    // Options to be passed to Mocha.
+    // See the full list at http://mochajs.org/
+    mochaOpts: {
+        ui: 'bdd',
+        timeout: 60000
+    },
+    //
+    // =====
+    // Hooks
+    // =====
+    // WebdriverIO provides several hooks you can use to interfere with the test process in order to enhance
+    // it and to build services around it. You can either apply a single function or an array of
+    // methods to it. If one of them returns with a promise, WebdriverIO will wait until that promise got
+    // resolved to continue.
+    /**
+     * Gets executed once before all workers get launched.
+     * @param {Object} config wdio configuration object
+     * @param {Array.<Object>} capabilities list of capabilities details
+     */
+    onPrepare: function (config) {
+      const webpackConfig = webpackConfigFn(),
+          server = new WebpackDevServer(webpack(webpackConfig), webpackConfig.devServer);
+
+      // save the server so we can shut it down in onComplete
+      config.webpackServer = server;
+
+      console.log('Starting WebpackDevServer');
+
+      return new Promise(function(resolve, reject) {
+        server.listen(webpackConfig.devServer.port, webpackConfig.devServer.host, function(err) {
+          if (err) {
+            reject(err);
+          }
+          else {
+            console.log('WebpackDevServer started successfully on ' +
+                `http://${webpackConfig.devServer.host}:${webpackConfig.devServer.port}`);
+
+            resolve();
+          }
+        });
+      });
+    },
+    /**
+     * Gets executed before a worker process is spawned and can be used to initialise specific service
+     * for that worker as well as modify runtime environments in an async fashion.
+     * @param  {String} cid      capability id (e.g 0-0)
+     * @param  {[type]} caps     object containing capabilities for session that will be spawn in the worker
+     * @param  {[type]} specs    specs to be run in the worker process
+     * @param  {[type]} args     object that will be merged with the main configuration once worker is initialised
+     * @param  {[type]} execArgv list of string arguments passed to the worker process
+     */
+    // onWorkerStart: function (cid, caps, specs, args, execArgv) {
```

Comment:
I assume that you copied this config file from a template and that all of these hooks that aren't used right now will be gone on cleanup, right?

Reply:
> I assume that you copied this config file from a template and that all of these hooks that aren't used right now will be gone on cleanup, right?

Well, generated by an init command, but close enough.  I wasn't planning to remove all the commented out stuff.  Do you think I should?

Reply:
> > I assume that you copied this config file from a template and that all of these hooks that aren't used right now will be gone on cleanup, right?
> 
> Well, generated by an init command, but close enough. I wasn't planning to remove all the commented out stuff. Do you think I should?

I don't feel strongly about it, though it does improve readability a bit. 
I will ultimately leave the decision to you.